### ISSUE TITLE
Enable SQL problem

### ISSUE THREAD

**Issue by BlancoConsulting** [halcyon-tech/vscode-ibmi#1559]
When "Enable SQL" is checked the source files and members do not appear, and the library description says "*** NOT FOUND ***".

When See attached PDF.

[EnableSQLProblem.pdf](https://github.com/halcyon-tech/vscode-ibmi/files/12742411/EnableSQLProblem.pdf)

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.3.1|
|Visual Studio Code version|1.82.2|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
Beautify (beautify): 1.5.0
COBOL (cobol): 9.9.21
Code Spell Checker (code-spell-checker): 3.0.1
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.3.2
Comment Anchors (comment-anchors): 1.10.3
Db2 for IBM i (vscode-db2i): 0.5.0
Dev Containers (remote-containers): 0.309.0
ESLint (vscode-eslint): 2.4.2
Emmet (emmet): 1.0.0
Eradani Extension (ec-generators): 0.9.11
Error Lens (errorlens): 3.14.0
Extension Authoring (extension-editing): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
Git Graph (git-graph): 1.30.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
GitHub Copilot (copilot): 1.117.443
GitHub Copilot Chat (copilot-chat): 0.7.1
GitLens — Git supercharged (gitlens): 14.3.0
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
JSON Language Features (json-language-features): 1.0.0
Live Server (LiveServer): 5.7.9
Markdown All in One (markdown-all-in-one): 3.5.1
Markdown Math (markdown-math): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Microsoft Account (microsoft-authentication): 0.0.1
NPM support for VS Code (npm): 1.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Remote - Tunnels (remote-server): 1.4.3
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL (remote-wsl): 0.81.5
WSL: Recommender (remote-wsl-recommender): 0.0.19
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|n/a|
|Tech Refresh|n/a|
|CCSID|?|
|SQL|Disabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|
|-|-|-|-|
|bash|attr|GENCMDXML.PGM|QZDFMDB2.PGM|
|git|iconv|GETNEWLIBL.PGM||
|grep|setccsid|||
|ls||||
|md5sum||||
|sort||||
|stat||||
|tar||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=RAMIRO
CURLIB=RAMIRO
HOME=/home/RAMIRO
HOST=Blue
LIBLS=QGPL
LOGIN=ramiro
LOGNAME=ramiro
MAIL=/var/spool/mail/ramiro
OLDPWD=/home/ramiro
PASE_USRGRP_LIMITED=N
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PWD=/home/RAMIRO
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=10.231.45.253 54833 22
SSH_CONNECTION=10.231.45.253 54833 172.20.214.6 22
TZ=<EST>5<EDT>,M3.2.0,M11.1.0
USER=ramiro
USERNAME=ramiro
_=/QOpenSys/usr/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "system \"CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')\"",
    "code": 255,
    "stderr": "CPF2111: Library ILEDITOR already exists.",
    "cwd": "/home/RAMIRO"
  },
  {
    "command": "system \"DLTOBJ OBJ(ILEDITOR/O_*) OBJTYPE(*FILE)\"",
    "code": 255,
    "stderr": "CPF2117: 0 objects type *FILE deleted. 12 objects not deleted.",
    "cwd": "/home/RAMIRO"
  },
  {
    "command": "ls undefined",
    "code": 2,
    "stderr": "undefined not found",
    "cwd": "/home/RAMIRO"
  }
]
```
</details>

**Comment by worksofliam** [comment]
Please share the contents of your Code for IBM i output channel after the list doesn't appear.

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3708366/178739532-03c8ad30-84de-4698-9904-65938f2ef651.png">

**Comment by BlancoConsulting** [comment]
Here is the Code for IBM i output after checked "Enable SQL" and list doesn't appear:

.: echo "This should be the only text!"
{
    "code": 0,
    "signal": null,
    "stdout": "This should be the only text!",
    "stderr": ""
}

.: echo $HOME && cd && test -w $HOME
{
    "code": 0,
    "signal": null,
    "stdout": "/home/RAMIRO",
    "stderr": ""
}

/home/RAMIRO: /QOpenSys/usr/bin/qsh
liblist
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/RAMIRO: system "CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')"
{
    "code": 255,
    "signal": null,
    "stdout": "",
    "stderr": "CPF2111: Library ILEDITOR already exists."
}

/home/RAMIRO: [ -d "/tmp" ]
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/home/RAMIRO: system "DLTOBJ OBJ(ILEDITOR/O_*) OBJTYPE(*FILE)"
/home/RAMIRO: rm -f /tmp/vscodetemp*
/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
select CHARACTER_CODE_SET_ID from table( QSYS2.QSYUSRINFO( USERNAME => upper('ramiro') ) )
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

{
    "code": 255,
    "signal": null,
    "stdout": "",
    "stderr": "CPF2117: 0 objects type *FILE deleted. 12 objects not deleted."
}

{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
with VARIANTS ( HASH, AT, DOLLARSIGN ) as (  values ( cast( x'7B' as varchar(1) )         , cast( x'7C' as varchar(1) )         , cast( x'5B' as varchar(1) ) ))select HASH concat AT concat DOLLARSIGN as LOCAL  from VARIANTS; 
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"

        select os.OBJNAME as ODOBNM
             , coalesce(os.OBJTEXT, '') as ODOBTX
             , os.OBJATTRIBUTE as ODOBAT
          from table( SYSTOOLS.SPLIT( INPUT_LIST => 'RAMIRO,QGPL', DELIMITER => ',' ) ) libs
             , table( QSYS2.OBJECT_STATISTICS( OBJECT_SCHEMA => 'QSYS', OBJTYPELIST => '*LIB', OBJECT_NAME => libs.ELEMENT ) ) os
      
/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
select OS_VERSION || '.' || OS_RELEASE as VERSION from sysibmadm.env_sys_info
/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
select max(ptf_group_level) as HIGHEST_DB2_PTF_GROUP_LEVEL from qsys2.group_ptf_info where PTF_GROUP_DESCRIPTION like 'DB2 FOR IBM I%' and ptf_group_status = 'INSTALLED';
/home/RAMIRO: ls undefined
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

{
    "code": 2,
    "signal": null,
    "stdout": "",
    "stderr": "undefined not found"
}

{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"

        select os.OBJNAME as ODOBNM
             , coalesce(os.OBJTEXT, '') as ODOBTX
             , os.OBJATTRIBUTE as ODOBAT
          from table( SYSTOOLS.SPLIT( INPUT_LIST => 'RAMIRO,QGPL', DELIMITER => ',' ) ) libs
             , table( QSYS2.OBJECT_STATISTICS( OBJECT_SCHEMA => 'QSYS', OBJTYPELIST => '*LIB', OBJECT_NAME => libs.ELEMENT ) ) os
      
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"

        select os.OBJNAME as ODOBNM
             , coalesce(os.OBJTEXT, '') as ODOBTX
             , os.OBJATTRIBUTE as ODOBAT
          from table( SYSTOOLS.SPLIT( INPUT_LIST => 'RAMIRO,QGPL', DELIMITER => ',' ) ) libs
             , table( QSYS2.OBJECT_STATISTICS( OBJECT_SCHEMA => 'QSYS', OBJTYPELIST => '*LIB', OBJECT_NAME => libs.ELEMENT ) ) os
      
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

/home/RAMIRO: system "DSPFD FILE(RAMIRO/Q*) TYPE(*ATR) FILEATR(*PF) OUTPUT(*OUTFILE) OUTFILE(ILEDITOR/O_weKbFz4b)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPF9861: Output file O_WEKBFZ4B created in library ILEDITOR.\nCPF9862: Member O_WEKBFZ4B added to output file O_WEKBFZ4B in library ILEDITOR.\nCPF3030: 7 records added to member O_WEKBFZ4B in file O_WEKBFZ4B in ILEDITOR.",
    "stderr": ""
}

/home/RAMIRO: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
SELECT * FROM ILEDITOR.O_weKbFz4b
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   **** Generated from Blue *****",
    "stderr": ""
}

.: system "DLTOBJ OBJ(ILEDITOR/O_weKbFz4b) OBJTYPE(*FILE)"
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

**Comment by BlancoConsulting** [comment]
It's generated by the system for identification purpose.

**Comment by BlancoConsulting** [comment]
Hi Barry, I have another issue open "Db2 for i update process failing #1558" which is of higher priority.  The workaround of this issue "Enable SQL problem #1559" is to uncheck it.  Although, it would be nice to find a solution.

**Comment by worksofliam** [comment]
Well that being printed is likely the issue to be honest. This is also likely to be a similar issue for the Db2 extension.

It is usually related to authority. Ensure you have write/read access to your home directory (usually `/home/USER`)

**Comment by BlancoConsulting** [comment]
I do have authority to /home/RAMIRO

**Comment by BlancoConsulting** [comment]
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/77461463/94f68e4d-790a-4a71-a37e-edbefeeb57cb)

**Comment by BlancoConsulting** [comment]
Barry, other users having the same problem?  Is there any hope for a solution?

**Comment by worksofliam** [comment]
@BlancoConsulting Your situation is slightly unique but others have had similar issues.

1. `"stdout": "DB2>\n **** Generated from Blue *****",` - this appearing in your standard will cause things not to work. Sounded like you knew where it was coming from so I will leave that one to you.
2. I can also see from the output that no output is coming back from SQL execution. See 'No results from SQL execution' [here](https://halcyon-tech.github.io/docs/#/pages/tips/setup?id=no-results-from-sql-execution).

**Comment by BlancoConsulting** [comment]
This fixed the issue #1559:

"See 'No results from SQL execution' [here](https://halcyon-tech.github.io/docs/#/pages/tips/setup?id=no-results-from-sql-execution)."

Now, the issue "Db2 for i update process failing #1558" is not fixed.

I am closing this issue.
